org.adempiere.base/src/org/compiere/model/MInOut.java

```diff
@@ -1037,7 +1037,34 @@ protected boolean beforeSave (boolean newRecord)
             MDocType docType = MDocType.get(getCtx(), rma.getC_DocType_ID());
             setC_DocType_ID(docType.getC_DocTypeShipment_ID());
         }
+        
+        // Set MovementType
+        if(newRecord || (get_ValueDifference("C_DocType") != null)) {
+			MDocType docType = new MDocType(getCtx(), getC_DocType_ID(), get_TrxName());
+			String DocBaseType = docType.getDocBaseType();
+			Boolean IsSOTrx = docType.isSOTrx();
+			
+			if (DocBaseType.equals("MMS"))					//	Material Shipments
+				
+				{
+						if (IsSOTrx)
+							setMovementType("C-");				// Customer Shipments
+						else
+							setMovementType("V-");				// Vendor Return
 
+				}
+				
+			else if (DocBaseType.equals("MMR"))				//	Material Receipts
+			    
+				{
+						if (IsSOTrx)
+							setMovementType("C+");				 // Customer Return
+						else
+							setMovementType("V+");				 // Vendor Receipts
+				}				
+				
+		}
+        
```

Comment:
I think this adapted from CalloutInOut.doctype. To ease future maintenance, it would be much better to refactor CalloutInOut.doctype and share the logic here (obviously, the logic of both must be the same).